deckhouse-controller/pkg/apis/deckhouse.io/validation/validate_module_config.go

```diff
@@ -31,25 +33,104 @@ import (
 	d8config "github.com/deckhouse/deckhouse/go_lib/deckhouse-config"
 )
 
+type AnnotationsOnly struct {
+	ObjectMeta `json:"metadata,omitempty"`
+}
+
+type ObjectMeta struct {
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
 // moduleConfigValidationHandler validations for ModuleConfig creation
-func moduleConfigValidationHandler() http.Handler {
-	vf := kwhvalidating.ValidatorFunc(func(_ context.Context, review *model.AdmissionReview, obj metav1.Object) (result *kwhvalidating.ValidatorResult, err error) {
+func moduleConfigValidationHandler(moduleStorage ModuleStorage, metricStorage *metric_storage.MetricStorage) http.Handler {
+	vf := kwhvalidating.ValidatorFunc(func(_ context.Context, review *kwhmodel.AdmissionReview, obj metav1.Object) (result *kwhvalidating.ValidatorResult, err error) {
+		var (
+			cfg = new(v1alpha1.ModuleConfig)
+			ok  bool
+		)
+
 		switch review.Operation {
 		case kwhmodel.OperationDelete:
-			// Always allow deletion.
-			return allowResult("")
-
+			{
+				cfg, ok = obj.(*v1alpha1.ModuleConfig)
+				if !ok {
+					return nil, fmt.Errorf("expect ModuleConfig as unstructured, got %T", obj)
+				}
+
+				// if we have no annotation and module enabled
+				//
+				// we check module
+				// we check confirmation restriction and confirmation message
+				_, ok = cfg.Annotations[v1alpha1.AllowDisableAnnotation]
+				if !ok && *cfg.Spec.Enabled {
+					module, err := moduleStorage.GetModuleByName(obj.GetName())
+					if err != nil {
+						return rejectResult(fmt.Sprintf("Module '%s' not registered in deckhouse", obj.GetName()))
```

Comment:
Oh, we don't need this. Otherwise we can't delete a misfiled ModuleConfig


```
$ deckhouse-controller module enable foobar
Module foobar enabled

$ kubectl get mc foobar
NAME     ENABLED   VERSION   AGE   MESSAGE
foobar   true                15s   Ignored: unknown module name

$ deckhouse-controller module disable foobar
deckhouse-controller: error: disable module failed: update ModuleConfig/foobar: admission webhook "module-configs.deckhouse-webhook.deckhouse.io" denied the request: Module 'foobar' not registered in deckhouse, try --help

$ kubectl delete mc foobar
Error from server: admission webhook "module-configs.deckhouse-webhook.deckhouse.io" denied the request: Module 'foobar' not registered in deckhouse
```

Reply:
I think we can delete unknown module without any further check

Reply:
done